plugins/inventory/hrobot.py

```diff
@@ -0,0 +1,180 @@
+    separator: ""
+"""
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.module_utils.urls import open_url
+from ansible.utils.display import Display
+from ansible.errors import AnsibleError
+
+import json
+
+from ansible_collections.community.hrobot.plugins.module_utils.robot import (
+    BASE_URL
+)
+
+display = Display()
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'hrobot'
+
+    def verify_file(self, path):
+        ''' return true/false if this is possibly a valid file for this plugin to consume '''
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            # base class verifies that file exists and is readable by current user
+            if path.endswith(('hrobot.yaml', 'hrobot.yml')):
+                valid = True
+        display.debug("hrobot inventory filename must end with 'hrobot.yml' or 'hrobot.yaml'")
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+        servers = {}
+        config = self._read_config_data(path)
+        self.load_cache_plugin()
+        cache_key = self.get_cache_key(path)
+
+        # cache may be True or False at this point to indicate if the inventory is being refreshed
+        # get the user's cache option too to see if we should save the cache if it is changing
+        user_cache_setting = self.get_option('cache')
+
+        # read if the user has caching enabled and the cache isn't being refreshed
+        attempt_to_read_cache = user_cache_setting and cache
+        # update if the user has caching enabled and the cache is being refreshed; update this value to True if the cache has expired below
+        cache_needs_update = user_cache_setting and not cache
+
+        # attempt to read the cache if inventory isn't being refreshed and the user has caching enabled
+        if attempt_to_read_cache:
+            try:
+                servers = self._cache[cache_key]
+            except KeyError:
+                # This occurs if the cache_key is not in the cache or if the cache_key expired, so the cache needs to be updated
+                cache_needs_update = True
+        else:
+            servers = self.get_servers()
+
+        if cache_needs_update:
+            servers = self.get_servers()
+
+            # set the cache
+            self._cache[cache_key] = servers
+
+        self.populate(servers)
+
+    def populate(self, servers):
+        filters = self.get_option("filters")
+        for server in servers:
+            s = server["server"]
+
+            matched = self.filter(s, filters)
+            if matched:
+                self.inventory.add_host(s["server_name"])
+                self.inventory.set_variable(s["server_name"], 'ansible_host', s['server_ip'])
+                self.inventory.set_variable(s["server_name"], 'server_ip', s['server_ip'])
```

Comment:
I think all non-ansible variable names should be prefixed with `hrobot_` or something like that. Things like `server_ip` are way too generic.

Reply:
I've checked few plugins, basically all set 1:1. I'm thinking about maintain all fields like
```
                self.inventory.add_host(s["server_name"])
                for hostvar, hostval in s.items():
                    self.inventory.set_variable(s["server_name"], hostvar, hostval)
```

But if you want to have a prefix for each like `hrobot_` I could fix with concatenation 

```
                self.inventory.add_host(s["server_name"])
                for hostvar, hostval in s.items():
                    self.inventory.set_variable(s["server_name"], "{0}_{1}".format("hborot",hostvar), hostval)
```

Reply:
Just because other plugins do that doesn't mean it's a good idea :) A prefix would be a lot better IMO.

Reply:
ok ok. thanks!

Reply:
@felixfontein just thought about mismatch and confusion when specifying a filter like 
```
plugin: hrobot
filters:
  status: ready
  traffic: unlimited
```
^ goes https://robot.your-server.de/doc/webservice/en.html#get-server definition, but then user has to reference the same with `hrobot_` prefix.

Reply:
You could adjust `filter` to also require the prefixes (or allow both variants).